apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/BufferOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// Types of buffers.
+    /// </summary>
+    public enum BufferType
+    {
+        /// <summary>No buffer will be used and the base consumer will be returned.</summary>
+        None,
+
+        /// <summary>A sized buffer will be used to wrap the base consumer.</summary>
+        Sized,
+
+        /// <summary>A timed buffer will be used to wrap the base consumer.</summary>
+        Timed
+    }
+
+    /// <summary>
+    /// Options that define a buffer.
+    /// </summary>
+    public sealed class BufferOptions
+    {
+        /// <summary>
+        /// Singleton for a <see cref= "BufferOptions" /> instance that describes a 
+        /// <see cref="BufferType.None"/> type of buffer.
+        /// </summary>
+        private static readonly BufferOptions NoBufferInstance = new BufferOptions(BufferType.None);
```

Comment:
Nit that can easily be fixed later - it's private, after all. As this is a static field (not a const), this should be `s_noBufferInstance`.